core/src/main/java/io/opencensus/stats/Aggregate.java

```diff
@@ -0,0 +1,254 @@
+
+    AggregateSum() {
+    }
+
+    static AggregateSum create(double sum) {
+      return new AutoValue_Aggregate_AggregateSum(sum);
+    }
+
+    /**
+     * Returns the aggregated sum.
+     *
+     * @return the aggregated sum.
+     */
+    public abstract double get();
+
+    @Override
+    public final <T> T match(
+        Function<? super AggregateSum, T> p0,
+        Function<? super AggregateCount, T> p1,
+        Function<? super AggregateHistogram, T> p2,
+        Function<? super AggregateRange, T> p3,
+        Function<? super AggregateMean, T> p4,
+        Function<? super AggregateStdDev, T> p5) {
+      return p0.apply(this);
+    }
+  }
+
+  /** The count value of aggregated {@code MeasureValue}s. */
+  @Immutable
+  @AutoValue
+  public abstract static class AggregateCount extends Aggregate {
+
+    AggregateCount() {
+    }
+
+    static AggregateCount create(long count) {
+      return new AutoValue_Aggregate_AggregateCount(count);
+    }
+
+    /**
+     * Returns the aggregated count.
+     *
+     * @return the aggregated count.
+     */
+    public abstract long get();
+
+    @Override
+    public final <T> T match(
+        Function<? super AggregateSum, T> p0,
+        Function<? super AggregateCount, T> p1,
+        Function<? super AggregateHistogram, T> p2,
+        Function<? super AggregateRange, T> p3,
+        Function<? super AggregateMean, T> p4,
+        Function<? super AggregateStdDev, T> p5) {
+      return p1.apply(this);
+    }
+  }
+
+  /** The histogram of aggregated {@code MeasureValue}s. */
+  @Immutable
+  @AutoValue
+  public abstract static class AggregateHistogram extends Aggregate {
+
+    AggregateHistogram() {
+    }
+
+    static AggregateHistogram create(long... bucketCounts) {
+      checkNotNull(bucketCounts, "bucket counts should not be null.");
+      List<Long> boxedBucketCounts = new ArrayList<Long>();
+      for (long bucketCount : bucketCounts) {
+        boxedBucketCounts.add(bucketCount);
+      }
+      return new AutoValue_Aggregate_AggregateHistogram(
+          Collections.unmodifiableList(boxedBucketCounts));
+    }
+
+    /**
+     * Returns the aggregated bucket counts. The returned list is immutable, trying to update it
+     * will throw an {@code UnsupportedOperationException}.
+     *
+     * @return the aggregated bucket counts.
+     */
+    public abstract List<Long> get();
+
+    @Override
+    public final <T> T match(
+        Function<? super AggregateSum, T> p0,
+        Function<? super AggregateCount, T> p1,
+        Function<? super AggregateHistogram, T> p2,
+        Function<? super AggregateRange, T> p3,
+        Function<? super AggregateMean, T> p4,
+        Function<? super AggregateStdDev, T> p5) {
+      return p2.apply(this);
+    }
+  }
+
+  /** The range of aggregated {@code MeasureValue}s. */
+  @Immutable
+  @AutoValue
+  public abstract static class AggregateRange extends Aggregate {
```

Comment:
This is from our previous implementation for [DistributionAggregate](https://github.com/census-instrumentation/opencensus-java/blob/master/core/src/main/java/io/opencensus/stats/DistributionAggregate.java#L115), its usage is like `distributionAggregate.getRange().getMin()` or `.getMax()`. I've updated the new `Aggregate` so that `Range` will be just a subclass of `Aggregate`, and has fields min and max.